bsdlib/README.rst

```diff
@@ -3,9 +3,131 @@
+* 1 GNSS/AGPS socket
+* 1 Modem DFU socket
+* 10 PDN sockets
+
+Overview
+********
+
+.. figure:: doc/images/bsd_lib_overview.svg
+   :alt: BSD library overview diagram
+
+   BSD library overview diagram
+
+The BSD library is the primary interface to operate the nRF9160 modem in order to establish LTE-M, NBIOT, and GNSS connections.
+The library abstracts away the details on memory management and synchronization leaving the application to focus on writing the application using standardized API calls.
+
+The current implementation of the BSD library supports the following protocols:
+
+* TCP (client/server)
+* UDP
+* RAW sockets
+* DTLS 1.2 client
+* TLS 1.2 client
+* AT commands
+* GNSS
+* AGPS
+* PDN management
+* Modem DFU
+
+.. note::
+   The DNS address management protocol is not yet implemented.
+
+The library implements the following standard BSD functions:
+   
+* socket
+* close
+* bind
+* listen
+* accept
+* connect
+* recvfrom
+* recv
+* read
+* sendto
+* send
+* write
+* getsockopt
+* setsockopt
+* fcntl
+* poll
+* inet_pton
+* inet_ntop
+* getaddrinfo
+* freeaddrinfo
+
+.. note::
+   The ``select`` function is not yet implemented.
+
+
+Library internals
+*****************
+
+The purpose of the BSD library is to communicate with the nRF9160 modem firmware.
+This is done by sending events and getting responses over the Inter Processor Communication peripheral (IPC).
+To facilitate data transfer between the application and the modem, payload data is set in the shared RAM.
+
+The modem contains a full IP and DTLS/TLS stack as well as GNSS.
+The library internally communicates over a Remote Procedure Call (RPC) protocol using the IPC and the shared RAM to utilize the modules inside the modem.
+The library provides address families, types, and protocols that dispatch the protocols to the right backend in the modem and thus communicates with this backend.
+
+The BSD library aims to provide a standard programming interface on top of all of these backends so that the application developer does not have to manage all intricate details of the low-level parameters passed back and forth between the application and the modem.
+
+The figure below shows a simplified BSD library architecture.
+
+.. figure:: doc/images/bsd_lib_architecture.svg
+   :alt: BSD library architecture diagram
+
+   BSD library architecture diagram
+
+Porting flexibility
+*******************
+
+The BSD library has been designed in such way that you can port it to any OS.
+Therefore, procedures that would normally require some kind of OS interaction, like going into sleep and IRQ reprioritization, have been moved out of the library to an open :file:`c` file in which OS-specific integration  can be implemented.
+For more information, see :ref:`bsdlib_ug_porting`.
+
+Limitations
+***********
+
+GNSS, AGPS, and Modem DFU are currently not part of the generic sockets.
+Therefore, operating on them using ``poll`` is not possible yet.
```

Comment:
There are also several other function calls that are not supported; see my comment above.